File: src/main/java/com/example/first/ServerMain.java
package com.example.first;

import java.io.IOException;
import java.io.InputStream;
import java.net.ServerSocket;
import java.net.Socket;
import java.util.Scanner;

public class ServerMain {

    public static void main(String[] args) throws IOException {
        System.out.println("Server init!");
        final ServerSocket server = new ServerSocket(8080);

        final Socket socket = server.accept();
        System.out.println("Client connected!");

        final InputStream inputStream = socket.getInputStream();

        final Scanner scanner = new Scanner(inputStream);

        System.out.println(scanner.nextLine());

        scanner.close();

        inputStream.close();

        socket.close();

    }
}

File: src/main/kotlin/com/example/server/ServerMain.kt
package com.example.server

import java.net.ServerSocket
import kotlin.concurrent.thread


private const val PORT = 8080

fun main() {
    println("Server init ...")
    val server = ServerSocket(PORT) // final ServerSocket server = new ServerSocket(8080)
    println("Server inited! on $PORT")
    val socket = server.accept()
    println("Client connected!")

    thread {
        val inputStream = socket.getInputStream()
        val reader = inputStream.bufferedReader()
        while (true) {
            println(reader.readLine())
        }
    }

    thread {
        val outputStream = socket.getOutputStream()
        val writer = outputStream.bufferedWriter()

        while (true) {
            writer.write(readln())
            writer.newLine()
            writer.flush()
        }
    }


}